### DISCUSSION TITLE
Issue with Vite

### DISCUSSION THREAD

**Discussion by Sarimarcus** [SjorsO/deploy-laravel#2] [Q&A]
Hello, first of all thanks for sharing this deployment script.
I have an issue with Vite, I'm having this issue:

`[previous exception] [object] (Illuminate\\Foundation\\ViteManifestNotFoundException(code: 0): Vite manifest not found at: /var/www/ai-wp-factory/releases/2/public/build/manifest.json at /var/www/ai-wp-factory/releases/2/vendor/laravel/framework/src/Illuminate/Foundation/Vite.ph>`

Actually, the file is there:
```
[olivier@xxx:/var/www/ai-wp-factory/releases/2/public/build/.vite]  ± ls -al
total 12
drwxr-xr-x 2 www-data www-data 4096 Oct 13 19:42 .
drwxr-xr-x 4 www-data www-data 4096 Oct 13 19:42 ..
-rw-r--r-- 1 www-data www-data  255 Oct 13 19:42 manifest.json
```

Any idea about what's wrong? I tried to clear all the cache, but it's not working :/

Thanks for your help

**Comment by SjorsO** [comment]
The exception says it's looking for the manifest at `public/build/manifest.json`, your manifest is at `public/build/.vite/manifest.json`. The `.vite` directory your manifest is in isn't the default directory, did you change it in the vite config file?

**Comment by Sarimarcus** [comment]
I've just saw that! 
Actually, no: 

```
import { defineConfig } from "vite";
import laravel from "laravel-vite-plugin";

export default defineConfig({
    plugins: [
        laravel({
            input: ["resources/css/app.css", "resources/js/app.js"],
            refresh: true,
        }),
    ],
});
```

But I have issues with Vite, I'm having errors when running: `npm ci`: 

```
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: laravel-vite-plugin@0.8.1
npm error Found: vite@5.1.8
npm error node_modules/vite
npm error   dev vite@"^5.1.8" from the root project
npm error
npm error Could not resolve dependency:
npm error peer vite@"^3.0.0 || ^4.0.0" from laravel-vite-plugin@0.8.1
npm error node_modules/laravel-vite-plugin
npm error   dev laravel-vite-plugin@"^0.8.0" from the root project
npm error
npm error Conflicting peer dependency: vite@4.5.5
npm error node_modules/vite
npm error   peer vite@"^3.0.0 || ^4.0.0" from laravel-vite-plugin@0.8.1
npm error   node_modules/laravel-vite-plugin
npm error     dev laravel-vite-plugin@"^0.8.0" from the root project
```

I had to use : `npm ci --legacy-peer-deps` to bypass the error. I don't know if it's related.

**Comment by SjorsO** [comment]
I don't think your issue is related to the deployment script. 

As a solution you can configure Laravel to look for the manifest file in the `.vite` directory. You could also add a `cp public/build/.vite/manifest.json public/build/manifest.json` to the deployment script after the `npm run build` line

**Comment by Sarimarcus** [comment]
Thank you for the information. It's working now!

**Comment by nysirpayne** [comment]
308910d2c705047bff858cbe84439fd2fb191acd
